src/xtgeo/grid3d/grid.py

```diff
@@ -1559,6 +1564,62 @@ def get_zoneprop_from_subgrids(self) -> NoReturn:
         """Make a XTGeo GridProperty instance for a Zone property subgrid index."""
         raise NotImplementedError("Not yet; todo")
 
+    def get_boundary_polygons(
+        self: Grid,
+        alpha_factor: Optional[float] = 1.0,
+        convex: Optional[bool] = False,
+        simplify: Optional[bool] = True,
+        filter_array: Optional[np.ndarray] = None,
+    ) -> Polygons:
+        """Extract boundary polygons from the grid cell centers.
+
+        A ``filter_array`` can be applied to extract boudaries around specific
+        parts of the grid e.g. a region or a zone.
+
+        The concavity and precision of the boundaries are controlled by the
+        ``alpha_factor``. A low ``alpha_factor`` makes more precise boundaries,
+        while a larger value makes more rough polygons.
+
+        Note that the ``alpha_factor`` is a multiplier (default value 1) on top
+        of an auto estimated value, derived from the maximum xinc and yinc from
+        the grid cells. Dependent on the regularity of the grid, tuning of the
+        alpha_factor (up/down) is sometimes necessary to get satisfactory results.
+
+        Args:
+            alpha_factor: An alpha multiplier, which controls the precision of the
+                bonudaries. A higher number will produce smoother and less accurate
+                polygons. Not applied if convex is set to True.
+            convex: The default is False, which means that a "concave hull" algorithm
+                is used. If convex is True, the alpha factor is overridden to a large
+                number, producing a 'convex' shape boundary instead.
+            simplify: If True, a simplification is done in order to reduce the number
+                of points in the polygons, where tolerance is 0.1. Another
+                alternative to True is to input a Dict on the form
+                ``{"tolerance": 2.0, "preserve_topology": True}``, cf. the
+                :func:`Polygons.simplify()` method. For details on e.g. tolerance, see
+                Shapely's simplify() method.
```

Comment:
I'm a bit skeptical of allowing `simplify` to be a boolean, dict, or None -- an optional boolean or optional dict perhaps, but not all three.

Is it an option to just let this function accept `tolerance: float = 0.1` and `preserve_topology: bool | None = None` as keyword arguments instead?

Reply:
I propose we keep as-is for the moment since we have the samme pattern when getting polygons from a surface. Will address proposed changes in a separate issue and PR. Although much applied *we think*, we need a deprecation mechanism

Reply:
Yes me and Jan discussed this a bit earlier that we should change it to something simpler and only keep the `tolerance` argument with default value 0.1 🙂  Was a bit unsure if I should do it on this PR or change it together with the surface method on a separate PR. I guess it doesn't really matter. Maybe just do the improved solution already for this one  @jcrivenaes and I create a PR tomorrow for the surface method where we add a deprecation in addition?